examples/simple-lambda.hs

```diff
@@ -7,26 +7,31 @@ import Control.Lens
 import qualified Data.ByteString.Lazy.Char8 as B
 import Data.Text (Text)
 import Stratosphere
+import Data.Aeson (object, Value(Array))
+import qualified Data.Vector as V
+import qualified Data.HashMap.Strict as HM
 
 main :: IO ()
 main = B.putStrLn $ encodeTemplate myTemplate
 
 myTemplate :: Template
 myTemplate =
   template
-  [ lambda ]
+  [ role, lambda ]
   & description ?~ "Lambda example"
   & formatVersion ?~ "2010-09-09"
 
 lambda :: Resource
 lambda =
-  resource "LambdaFunction" $
+  (resource "LambdaFunction" $
   LambdaFunctionProperties $
   lambdaFunction 
     lambdaCode 
     "index.handler" 
-    "arn:aws:iam::910653408535:role/MyLambdaBasicExecutionRole" 
+    "arn:aws:iam::910653408535:role/MyLambdaBasicExecutionRole"
     "nodejs4.3"
+  )
+  & dependsOn ?~ [ "IAMRole" ]
```

Comment:
Would be great to be able to reference the actual resource vs. the resource name string like i do here